backend/mem/file_test.go

```diff
@@ -0,0 +1,774 @@
+func (s *memFileTest) TestMoveToFile() {
+
+	expectedSlice := []byte("Hello World!")
+	newFile, err := s.fileSystem.NewFile("", "/samples/test.txt")
+	s.NoError(err, "Unexpected error creating file")
+	s.NoError(newFile.Touch(),"Unexpected error touching file")
+
+	_, err = s.testFile.Write(expectedSlice)
+	s.NoError(err, "Write failed unexpectedly")
+	s.NoError(s.testFile.Close(), "Unexpected close error")
+
+	//after this call, newFile and "s.testFile" will be deleted.
+	//we re-obtain the newFile pointer by calling it from our fsMap by giving it the (new) path and volume
+	err = s.testFile.MoveToFile(newFile)
+	s.NoError(err, "Move to file failed")
+	newFileSlice := make([]byte, len("Hello World!"))
+	newFile, err = s.fileSystem.NewFile("","/samples/test.txt")
+	s.NoError(err, "Unexpected creation error")
+
+	s.False(s.testFile.Exists())
+
+	_, err = newFile.Read(newFileSlice)
+	s.NoError(err, "Read unexpectedly failed")
+
+	s.Equal(string(expectedSlice), string(newFileSlice))
+	s.Equal("/samples/test.txt", newFile.Path())
+
+}
+
+//TestMoveToFile2 ensures that if a call is made on a files who don't share the same name, a new file is produced with the caller's name
+func (s *memFileTest) TestMoveToFile2() {
+
+	expectedSlice := []byte("Hello World!")
+	newFile, err := s.fileSystem.NewFile("", "/samples/diffName.txt")
+	s.NoError(err, "File creation was not successful so it does not exist")
+	//newPath := "/samples/test.txt" //this is the path used to retrieve the file after it has been moved
+	s.NoError(newFile.Touch(),"Unexpected error touching file")
+	_, err = s.testFile.Write(expectedSlice)
+	s.NoError(err, "Write failed unexpectedly")
+	s.NoError(s.testFile.Close(), "Unexpected close error")
+
+	//after this call, newFile and "s.testFile" will be deleted.
+	//we re-obtain the newFile by calling it from our fsMap by giving it the (new) path and volume
+	err = s.testFile.MoveToFile(newFile)
+	s.NoError(err, "Move to file failed")
+	newFileSlice := make([]byte, len("Hello World!"))
+	//newFile = s.fileSystem.fsMap[""][newPath].i.(*File)
+	s.False(s.testFile.Exists())
+
+	_, err = newFile.Read(newFileSlice)
+	s.NoError(err, "Read unexpectedly failed")
+	s.Equal(string(expectedSlice), string(newFileSlice))
+	s.Equal("/samples/diffName.txt", newFile.Path())
+
+}
+
+
+//TestWrite writes a string to a file and checks for success by comparing the number of bytes
+//written by "Write()" to the length of the slice it wrote from
+func (s *memFileTest) TestWrite() {
+	expectedText := "I'm fed up with this world" //-Tommy Wiseau
+	bSlice := []byte(expectedText)
+	length := len(bSlice)
+	num, err := s.testFile.Write(bSlice)
+	s.NoError(err, "Write did not work as expected")
+	s.EqualValues(length, num)
+
+}
+
+//TestWrite2 writes to a file, and reads from it without closing it by seeking to the start
+func (s *memFileTest) TestWrite2() {
```

Comment:
This function should have a more meaningful name and it should reflect what is actually being tested.